pkg/api/v1alpha1/tinkerbelltemplateconfig.go

```diff
@@ -81,11 +81,10 @@ func GetTinkerbellTemplateConfig(fileName string) (map[string]*TinkerbellTemplat
 			if err = yaml.UnmarshalStrict([]byte(c), &template); err == nil {
 				templates[template.Name] = &template
 				continue
+			} else {
+				return nil, fmt.Errorf("invalid template config content: %v", err)
 			}
```

Comment:
Prefer returning early over nesting.

```suggestion
			if err := yaml.UnmarshalStrict([]byte(c), &template); err != nil {
				return nil, fmt.Errorf("invalid template config content: %v", err)
			}
			
			templates[template.Name] = &template
```